workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncher.cs

```diff
@@ -0,0 +1,334 @@
+                }
+                EditorGUILayout.EndHorizontal();
+                EditorGUI.BeginDisabledGroup(runningListTask != null || runningStopTask != null || deploymentList.Count == 0);
+                if (GUILayout.Button("Stop Deployment"))
+                {
+                    runningStopTask = TriggerStopDeploymentAsync(deploymentList[selectedDeployment].Id);
+                }
+                EditorGUI.EndDisabledGroup();
+                EditorGUI.BeginDisabledGroup(runningListTask != null || runningStopTask != null);
+                if (GUILayout.Button("Refresh Deployments"))
+                {
+                    runningListTask = TriggerListDeploymentsAsync();
+                }
+                EditorGUI.EndDisabledGroup();
+            }
+
+            void Update()
+            {
+                if (runningStopTask != null && runningStopTask.IsCompleted)
+                {
+                    try
+                    {
+                        if (runningStopTask.Result)
+                        {
+                            deploymentList.RemoveAt(selectedDeployment);
+                            selectedDeployment = 0;
+                        }
+                        else
+                        {
+                            throw new Exception($"Failed to stop deployment {deploymentList[selectedDeployment].Name} with ID {deploymentList[selectedDeployment].Id}.");
+                        }
+                    }
+                    finally
+                    {
+                        runningStopTask = null;
+                        Repaint();
+                    }
+                }
+
+                if (runningListTask != null && runningListTask.IsCompleted)
+                {
+                    try
+                    {
+                        deploymentList.Clear();
+                        if (runningListTask.Result != null)
+                        {
+                            deploymentList = runningListTask.Result;
+                        }
+                        else
+                        {
+                            throw new Exception("Failed to refresh deployments list.");
+                        }
+                    }
+                    finally
+                    {
+                        runningListTask = null;
+                        Repaint();
+                    }
+                }
+            }
+
+            private async Task<bool> TriggerLaunchDeploymentAsync()
+            {
+                if (!CheckDependencies())
+                {
+                    return false;
+                }
+
+                assemblyName = assemblyName.Trim();
+                deploymentName = deploymentName.Trim();
+
+                if (string.IsNullOrEmpty(assemblyName))
+                {
+                    EditorUtility.DisplayDialog("Unable to start a deployment", "Please specify a valid assembly name.",
```

Comment:
Popups can be annoying, and rather than telling users after the fact, I suggest not letting them trigger this until their input data is in the correct state.
You could use a `HelpBox` (https://docs.unity3d.com/ScriptReference/EditorGUILayout.HelpBox.html) to show the user what's wrong and disable the launch button.

Reply:
Much better idea. Done.